### PR TITLE
Extract get event call from PullDhisApiDataSource, and refactor checkAllQuarantineSurveys to use domain entities.

### LINKED ISSUE #2016: Create a remote data source implementing ISurveyDataSource
Extract getEvent call from PullDhisApiDataSource to data source,  using ISurveyDataSource to call getEvents using SurveyFilter. 
Depends on #1904
depends on #2018

### PR DESCRIPTION
**Pull request by idelcano**
Closes #2016

### CODE REVIEW

**Comment by idelcano** [comment]
### :pushpin: References
* **Issue:** connect https://github.com/EyeSeeTea/malariapp/issues/2016

### :tophat: What is the goal?

Extract get event call from PullDhisApiDataSource, and refactor checkAllQuarantineSurveys to use domain entities.

### :memo: How is it being implemented?

I refactored the checkQuarantineMethod to use a UseCase.

I added a UpdateQuarantineSurveyStatusUseCase. 

This UseCase get the OrgUnit and Program Uids using a OrgUnitRepository, 
and create the necessary surveyFilters to check the surveys in quarantine using the repository.
The UseCase compares the local surveys against the pulled surveys to set as sent or as completed the quarantine surveys, and save them using the repository.

I added some creation methods and test it in surveyFilter and Survey.

I added a SurveyApiDatasource to get events from api data source, and remove all the old methods.

I have removed some SurveyDB queries and use entities to find the correct parameters to the filters.

I have add surveyQuarantineRepository, OrgUnitRepository and orgUnitDBMapper

I added a flag to avoid multiple Quarantine Use Cases instances because debugging i found this problem and it might make duplicates.

Updated:

finally the UpdateQuarantineSurveyStatusUseCase was removed moving that logic into the PushUseCase.
The logic of surveychecker has been simplified, asking to the server for the quarantine event uids, the orgunit/program/startdate/endate is not necessary anymore. 

### :boom: How can it be tested?

Note: I recommend use Cervical cancer program, if you will use leap server some other programs doesn't work (the user haven't permission to push).

- [ ] **Use Case 1:** 
Create a quarantine not saved in the server and check if is set as completed:
Login and mark as complete a survey.
Add a breakpoint in the line 124 of PushDataController.
Close the app.
Open the app and you will see in the log that survey is not sent and is set as complete, and push again.

- [ ] **Use Case 2:** 
Create a quarantine saved in the server and check if is set as sent:
Add a breakpoint in the line 154of PushDataController.
Evaluate the pushReport variable and set as null.
The survey will be set as quarantine, and if you see the logcat, the survey will be set as sent after quarantine check on next push.

### :floppy_disk: Requires DB migration?

- [x] Nope, we can just merge this branch.
- [ ] Yes, but we need to apply it before merging this branch.
- [ ] Yes, it's already applied.

### :art: UI changes?

- [x] Nope, the UI remains as beautiful as it was before!
- [ ] Yeap, here you have some screenshots-

**Review by xurxodev — CHANGES_REQUESTED** [review]
@idelcano please resolve conflicts and refactor. Review the development branch because has architecture changes.

File: app/src/main/java/org/eyeseetea/malariacare/data/database/datasources/OrgUnitLocalDataSource.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.eyeseetea.malariacare.data.database.AppDatabase.orgUnitProgramRelationAlias;
+import static org.eyeseetea.malariacare.data.database.AppDatabase.orgUnitProgramRelationName;
+import static org.eyeseetea.malariacare.data.database.AppDatabase.programAlias;
+import static org.eyeseetea.malariacare.data.database.AppDatabase.programName;
+
+public class OrgUnitLocalDataSource implements IOrgUnitRepository {
```
Comment (xurxodev) [inline app/src/main/java/org/eyeseetea/malariacare/data/database/datasources/OrgUnitLocalDataSource.java]: There are several problems here:
This is a Datasource but implement IOrgUnitRepository
This DataSource is similar to OrgUnitRepository that implements IOrgUnitRepository also
Please refactor to we have Repository, localdatasource according to Clean Architecture with Read Policy

File: app/src/main/java/org/eyeseetea/malariacare/data/database/datasources/SurveyLocalDataSource.java [hunk 2]
```diff
@@ -204,4 +205,32 @@ private void loadValuesInSurveys(List<SurveyDB> surveyDBS) {
             }
         }
     }
+
+    public List<Survey> mapQuarantineSurveys(List<SurveyDB> surveysDB) {
```
Comment (xurxodev) [inline app/src/main/java/org/eyeseetea/malariacare/data/database/datasources/SurveyLocalDataSource.java]: this method is not used

File: app/src/main/java/org/eyeseetea/malariacare/domain/usecase/PushUseCase.java [hunk 3]
```diff
@@ -34,13 +37,19 @@
     private final IMainExecutor mMainExecutor;
     private IPushController mPushController;
     private Callback mCallback;
+    private Credentials credentials;
+    private Context context;
```
Comment (xurxodev) [inline app/src/main/java/org/eyeseetea/malariacare/domain/usecase/PushUseCase.java]: An use case should not have android dependencies

File: app/src/main/java/org/eyeseetea/malariacare/domain/usecase/UpdateQuarantineSurveyStatusUseCase.java [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+import org.eyeseetea.malariacare.domain.boundary.executors.IAsyncExecutor;
+import org.eyeseetea.malariacare.domain.entity.OrgUnit;
+import org.eyeseetea.malariacare.domain.entity.Survey;
+import org.eyeseetea.malariacare.domain.entity.SurveyStatus;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+public class UpdateQuarantineSurveyStatusUseCase implements UseCase{
```
Comment (xurxodev) [inline app/src/main/java/org/eyeseetea/malariacare/domain/usecase/UpdateQuarantineSurveyStatusUseCase.java]: This really is not a use case, the use case is push use case. An use case should can be execute independently and this use case only is executed during PushUseCase then is not a use case

**Review by xurxodev — CHANGES_REQUESTED** [review]
@idelcano I have requested you changes to review

**Comment by idelcano** [comment]
I added the first test using kotlin, i added the library following the IDe steps, but at the end the IDE is saying the version of the kotlin library in the  light sdk should be removed.

@xurxodev can you take a look to this error with gradle, kotlin, and the new sdk?

**Review by xurxodev — CHANGES_REQUESTED** [review]
@idelcano can you review failed Travis compilation?

**Pull request closed without merging**